third-party/rsutils/include/rsutils/number/running-average.h

```diff
@@ -0,0 +1,108 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <type_traits>
+#include <limits>
+#include <stdlib.h>  // size_t
+
+
+namespace rsutils {
+namespace number {
+
+
+// Compute the average of a set of numbers, one at a time, without overflow!
+// We adapt this to a signed integral T type, where we must start counting leftovers.
+// We also add some rounding.
+// 
+template< class T, typename Enable = void >
```

Comment:
Why is this declaration needed if you define the class later?

Reply:
This defines a generic template, mapping to nothing.
The later declarations specialize the template with specific arguments.